### PR TITLE
Dealing wth extra space

### PR DESCRIPTION
**Pull request by olusegz07**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)
https://tools.hmcts.net/jira/browse/EXUI-2631
Notes:
*
*
*

### CODE REVIEW

File: projects/rpx-xui-translation/src/lib/helpers/replace-placeholders/replace-placeholders.helper.ts [hunk 1]
```diff
@@ -16,29 +16,54 @@ export const replacePlaceholders = (input: string, replacements: Replacements) =
...
-  const matches = phrase.match(placeholderRegex) || [];
+  // Helper function to split text into spaces and content
+  const splitTextIntoComponents = (text: string): string[] => {
+    const components: string[] = [];
+    const trimmed = text.trim();
 
-  // Iterate through parts and matches to build the result array
-  parts.forEach((part, index) => {
-    const trimmedPart = part.trim();
+    if (text.startsWith(' ')) {
```
Comment (andywilkinshmcts) [inline projects/rpx-xui-translation/src/lib/helpers/replace-placeholders/replace-placeholders.helper.ts:31]: You need to handle multiple spaces - possibly OK for now, but just put a loop around it, and also for trailing space

File: projects/rpx-xui-translation/src/lib/rpx-translation.service.ts [hunk 2]
```diff
@@ -72,7 +72,7 @@ export class RpxTranslationService {
     ));
 
     return combineLatest(translatedComponents$).pipe(
-      map((values: string[]) => values.join(' ')) // Join the strings with a space or any separator
+      map((values: string[]) => values.join('')) // Join the strings with a space or any separator
```
Comment (andywilkinshmcts) [inline projects/rpx-xui-translation/src/lib/rpx-translation.service.ts:75]: Fix the comment to say - join the strings without spaces - separators are included in the values

File: projects/rpx-xui-translation/tsconfig.lib.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "types": [],
     "lib": [
       "dom",
-      "es2018"
+      "es2022"
```
Comment (andywilkinshmcts) [inline projects/rpx-xui-translation/tsconfig.lib.json:11]: Worried about this - do we need this change?

**Review by andywilkinshmcts — APPROVED** [review]
Kinda ok, but could be improved

**Review by andywilkinshmcts — APPROVED** [review]

**Review by Josh-HMCTS — APPROVED** [review]

**Pull request merged**